json-schema-generator/src/main/java/oracle/kubernetes/json/ObsoleteCheck.java

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2021, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.json;
+
+public interface ObsoleteCheck {
```

Comment:
A more Java-standard name for this interface would be something like `Obsoleteable`. A check is a process, whereas this describes an object that can have a particular property or behavior.